core/src/modules/mappingHelper.ts

```diff
@@ -9,7 +9,10 @@ export namespace MappingHelper {
     [remoteKey: string]: any;
   }
 
-  export async function getMapping(instance: Source | Destination) {
+  export async function getMapping(
+    instance: Source | Destination,
+    propertyColumn?: "key" | "id"
+  ) {
```

Comment:
The value here is the property key by default, but that's not what we want for the code config shape, so I made this more flexible. Also created a spec for it.